File: Tests/Pages/RemindPasswordView.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using Tests.Data;
using Tests.Extensions;

namespace Tests.Pages
{
    public class RemindPasswordView
    {
        private string _iframe = "remindPasswordView";
        private readonly IWebElement _openBtn;

        public RemindPasswordView(IWebElement openBtn) => _openBtn = openBtn;

        private IWebElement CloseBtn => WebDriver.Driver.FindElement(By.XPath("//button[text() = 'x']"));
        private IWebElement SendBtn => WebDriver.Driver.FindElement(By.XPath("//button[text() = 'Send']"));
        private IWebElement MessageLbl => WebDriver.Driver.FindElement(By.Id("message"));
        private IWebElement EmailFld => WebDriver.Driver.FindElement(By.Id("email"));

        public bool IsShown => WebDriver.Driver.FindElement(By.Id(_iframe)).Displayed;

        public void Open()
        {
            if (!IsShown)
            {
                _openBtn.Click();
            }
        }

        public void Close()
        {
            WebDriver.SwitchToFrame(_iframe);
            CloseBtn.Click();
            WebDriver.SwitchToParentFrame();
        }

        public (bool IsSuccessful, string Message) Send(string email)
        {
            WebDriver.SwitchToFrame(_iframe);
            EmailFld.SetText(email);
            SendBtn.Click();

            try
            {
                new WebDriverWait(WebDriver.Driver, TimeSpan.FromSeconds(Defaults.ImplicitWait))
                    .Until(_ => !string.IsNullOrEmpty(MessageLbl.Text) || WebDriver.Alert != null);
            }
            catch { /**/ }

            var result = (WebDriver.Alert is not null, WebDriver.Alert?.Text ?? MessageLbl.Text);

            WebDriver.Alert?.Accept();
            WebDriver.SwitchToParentFrame();

            return result;
        }
    }
}

File: Tests/src/test/java/pages/RemindPasswordView.java
package test.java.pages;

import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import test.java.utilities.Result;

public class RemindPasswordView extends PageWaits {
	private String remindPasswordView = "remindPasswordView";
	
	private WebElement getView() {
		driver.switchTo().parentFrame();
		return driver.findElement(By.id(remindPasswordView));
	}

	public RemindPasswordView(WebDriver driver) {
		super(driver);
	}

	public Result send(String email) {
		String resultMessage;
		By message = By.id("message");
		driver.switchTo().frame(remindPasswordView);
		driver.findElement(By.id("email")).sendKeys(email);
		driver.findElement(By.xpath("//button[text() = 'Send']")).click();

		waitForAlertOrElementVisible(message);

		boolean isAlertShown = isAlertShown();

		if (isAlertShown) {
			Alert alert = driver.switchTo().alert();
			resultMessage = alert.getText();
			alert.accept();
		} else {
			resultMessage = driver.findElement(message).getText();
		}

		driver.switchTo().parentFrame();
		
		return new Result(isAlertShown, resultMessage);
	}
	
	public void close() {
		driver.switchTo().frame(remindPasswordView);
		driver.findElement(By.xpath("//button[text() = 'x']")).click();
		driver.switchTo().parentFrame();
	}
	
	public RemindPasswordView open() {
		if (!isOpened()) {
			driver.findElement(By.id("remindBtn")).click();
		}
		
		return this;
	}
	
	public boolean isOpened() {
		driver.switchTo().parentFrame();
		return getView().isDisplayed();
	}
}